File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        res = []
        q = deque([root])

        while q:
            curr_end = None
            for _ in range(len(q)):
                curr_end = q.popleft()
                if curr_end.left:
                    q.append(curr_end.left)
                if curr_end.right:
                    q.append(curr_end.right)
            res.append(curr_end.val)
        
        return res

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        std::vector<std::vector<int>> result{};
        if (!root) return result;

        std::deque<TreeNode*> q{root};
        while (!q.empty()) {
            int level = q.size();
            std::vector<int> curr_level{};
            for (int i{}; i < level; ++i) {
                TreeNode* node = q.front();
                q.pop_front();
                curr_level.push_back(node->val);
                if (node->left) q.push_back(node->left);
                if (node->right) q.push_back(node->right);
            }
            result.push_back(std::move(curr_level));
        }
        return result;

    }
};
